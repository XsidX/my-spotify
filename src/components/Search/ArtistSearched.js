import React from 'react';
import { Link } from 'react-router-dom';
import propTypes from 'prop-types';

const ArtistSearched = ({ artist }) => {
  const { id, name, images, type } = artist;
  const image = images[0]?.url;

  return (
    <Link to={`/artist/${id}`} className="flex items-center gap-3">
      <div className="h-14 w-14 rounded-full overflow-hidden">
        <img src={image} alt={name} className="object-contain" />
      </div>
      <div className="flex flex-col justify-center">
        <h3 className="text-white text-sm">{name}</h3>
        <p className="text-gray-300 capitalize text-xs">{type}</p>
      </div>
    </Link>
  );
};

ArtistSearched.propTypes = {
  artist: propTypes.shape({
    id: propTypes.string.isRequired,
    name: propTypes.string.isRequired,
    images: propTypes.arrayOf(
      propTypes.shape({
        url: propTypes.string.isRequired,
      }),
    ).isRequired,
    type: propTypes.string.isRequired,
  }).isRequired,
};

export default ArtistSearched;
