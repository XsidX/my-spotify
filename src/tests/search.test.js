import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import artistsReducer from '../store/reducers/artists-slice';
import albumsReducer from '../store/reducers/albums-slice';
import uiReducer from '../store/reducers/ui-slice';
import ArtistSearched from '../components/Search/ArtistSearched';

const initialState = {
  artists: [],
  albums: [],
  ui: {
    toolBarIsIntersecting: false,
  },
};

const store = configureStore({
  reducer: {
    artists: artistsReducer,
    albums: albumsReducer,
    ui: uiReducer,
  },
  preloadedState: initialState,
});

describe('ArtistSearched', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ArtistSearched
              artist={{
                id: 'id',
                name: 'artist name',
                images: [{ url: 'url' }],
              }}
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
