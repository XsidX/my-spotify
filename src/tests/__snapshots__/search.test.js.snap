// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSearched renders correctly 1`] = `
<a
  className="flex items-center gap-3"
  href="/artist/id"
  onClick={[Function]}
>
  <div
    className="h-14 w-14 rounded-full overflow-hidden"
  >
    <img
      alt="artist name"
      className="object-contain"
      src="url"
    />
  </div>
  <div
    className="flex flex-col justify-center"
  >
    <h3
      className="text-white text-sm"
    >
      artist name
    </h3>
    <p
      className="text-gray-300 capitalize text-xs"
    />
  </div>
</a>
`;
